### PR TITLE
Implement `getMetadata` for remaining Passive scripts

### PR DESCRIPTION
**Pull request by ricekot**
Update the following scripts to implement the `getMetadata()` function:
- passive/Server Header Disclosure.js
- passive/SQL injection detection.js
- passive/Telerik Using Poor Crypto.js
- passive/Upload form discovery.js
- passive/X-Powered-By_header_checker.js

Also update evidence / otherInfo fields for:
- passive/clacks.js
- passive/CookieHTTPOnly.js

Part of #440.

### CODE REVIEW

File: passive/SQL injection detection.js [hunk 1]
```diff
@@ -1,383 +1,89 @@
...
+    const sqlImpl = sqlImplementations[i];
+    const alertTitle = `Information Disclosure - ${sqlImpl.name} error`;
+    const article = sqlImpl.name.match(/^(?:[IO]|MS|SQ|H)/) ? "An" : "A";
+    const alertDesc = `${article} ${sqlImpl.name} error was discovered in the HTTP response body.`;
+    if (sqlImpl.regex.test(body)) {
+      sqlImpl.regex.lastIndex = 0;
+      const found = [];
+      let comm;
+      while ((comm = sqlImpl.regex.exec(body))) {
+        found.push(comm[0]);
```
Comment (kingthorin) [inline passive/SQL injection detection.js:77]: Did this come from one of the comment scripts?

Comment (kingthorin) [reply passive/SQL injection detection.js:77]: If so, think it's worth renaming more meaningfully?

Comment (ricekot) [reply passive/SQL injection detection.js:77]: I combined all the individual checks into a single loop. Do you mean renaming the `found` variable? Maybe `foundSqlErrors`?

Comment (kingthorin) [reply passive/SQL injection detection.js:77]: Sorry I meant the “comm” array

File: passive/CookieHTTPOnly.js [hunk 2]
```diff
@@ -27,8 +27,13 @@ function scan(helper, msg, src) {
...
   if (cookies != null) {
     var re_noflag = /([Hh][Tt][Tt][Pp][Oo][Nn][Ll][Yy])/g;
-    if (!re_noflag.test(cookies)) {
-      helper.newAlert().setMessage(msg).setEvidence(cookies).raise();
+    if (!re_noflag.test(cookies.toString())) {
+      helper
+        .newAlert()
+        .setMessage(msg)
+        .setEvidence(cookies[0])
+        .setOtherInfo(`Other values: ${cookies.slice(1).toString()}`)
```
Comment (thc202) [inline passive/CookieHTTPOnly.js:35]: Should we do this only if there are more? Would avoid showing `Other values: `, more a nit though.

File: passive/SQL injection detection.js [hunk 3]
```diff
@@ -1,383 +1,89 @@
...
+name: Information Disclosure - SQL Error
+description: An SQL error was found in the HTTP response body.
+solution: >
+  Ensure proper sanitisation is done on the server side.
+risk: high
+confidence: medium
+cweId: 209  # CWE-209: Generation of Error Message Containing Sensitive Information
+wascId: 13  # WASC-13: Information Leakage
+status: alpha
+codeLink: https://github.com/zaproxy/community-scripts/blob/main/passive/Find%20Internal%20IPs.js
```
Comment (thc202) [inline passive/SQL injection detection.js:19]: Needs update. (We could add a test that checks the URL matches the path/filename, not asking for this PR.)

File: passive/CookieHTTPOnly.js (hunk 2 above)
Comment (ricekot) [reply passive/CookieHTTPOnly.js:35]: Yeah, I was also thinking that. Is it okay if I do that for all scripts in this PR?

Comment (thc202) [reply passive/CookieHTTPOnly.js:35]: Sounds good.

File: passive/SQL injection detection.js (hunk 1 above)
Comment (kingthorin) [reply passive/SQL injection detection.js]: Ugh I didn't realize how many times this had been copied. Anyway thanks for fixing this one. It seems that people have repeated use of comm[] all over the place :wink:

**Review by kingthorin — APPROVED** [review]

File: passive/Find Credit Cards.js [hunk 4]
```diff
@@ -65,6 +65,10 @@ function scan(helper, msg, src) {
           }
         }
         if (foundCard.length != 0) {
+          const otherInfo =
```
Comment (thc202) [inline passive/Find Credit Cards.js:68]: The `setOtherInfo(…)` call was not updated.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**